src/Apps/Collect2/Components/SeoProductsForCollection.tsx

```diff
@@ -0,0 +1,132 @@
+import { SeoProductsForCollection_artworks } from "__generated__/SeoProductsForCollection_artworks.graphql"
+import { Product } from "Components/v2/Seo/Product"
+import React from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+
+export interface SeoProductsForCollectionProps {
+  artworks: SeoProductsForCollection_artworks
+  collectionDescription: string
+  collectionURL: string
+  collectionName: string
+}
+
+// dont have anything about currency conversion in Artsy right now.
+// Just use the fixed ratio to show the approximated price.
+export function simpleConversionToUSD(price: number, currencyCode: string) {
+  if (currencyCode === "USD") {
+    return price
+  } else if (currencyCode === "GBP") {
+    return price * 1.28
+  } else if (currencyCode === "EUR") return price * 1.1
+}
+
+export const getMaxMinPrice = priceableArtworks => {
+  const priceRange = [null, null]
+  priceableArtworks.map(artwork => {
+    if (artwork.node.is_price_range && artwork.node.listPrice) {
+      const { maxPrice, minPrice } = artwork.node.listPrice
+      const convertedPrice = {
+        maxPrice: simpleConversionToUSD(maxPrice.major, maxPrice.currencyCode),
+        minPrice: simpleConversionToUSD(minPrice.major, minPrice.currencyCode),
+      }
+      if (!priceRange[1] || priceRange[1] < convertedPrice.maxPrice)
+        priceRange[1] = convertedPrice.maxPrice
+      if (!priceRange[0] || priceRange[0] > convertedPrice.minPrice)
+        priceRange[0] = convertedPrice.minPrice
+    } else if (artwork.node.listPrice && artwork.node.listPrice.major) {
+      const convertedPrice = simpleConversionToUSD(
+        artwork.node.listPrice.major,
+        artwork.node.listPrice.currencyCode
+      )
+      if (!priceRange[1] || convertedPrice > priceRange[1])
+        priceRange[1] = convertedPrice
+      if (!priceRange[0] || convertedPrice < priceRange[0])
+        priceRange[0] = convertedPrice
+    }
+  })
+  return priceRange
+}
+
+export const getFilteredArtworks = artworks_connection => {
+  const filtedWorks =
+    artworks_connection &&
+    artworks_connection.edges.filter(edge => {
+      return (
+        edge.node.listPrice &&
+        edge.node.sale_message !== "Contact For Price" &&
+        edge.node.sale_message !== "Sold"
+      )
+    })
+  return filtedWorks
+}
+
+export class SeoProducts extends React.Component<
+  SeoProductsForCollectionProps
+> {
+  render() {
+    const {
+      artworks: { artworks_connection },
+      collectionDescription,
+      collectionName,
+      collectionURL,
+    } = this.props
+
+    const priceAbleArtworks = getFilteredArtworks(artworks_connection)
+
+    return (
+      <>
+        <Product
+          data={{
+            name: collectionName,
+            description: collectionDescription,
+            url: collectionURL,
+            offers: {
+              "@type": "AggregateOffer",
+              lowPrice: getMaxMinPrice(priceAbleArtworks)[0],
```

Comment:
This is probably a very rare case but `lowPrice` is required so if it has an invalid value no `Product` metadata should be added to page.